### PR TITLE
Feature/moses/get auth user credentials

### PR DESCRIPTION
**Pull request by SapeleD3**

### CODE REVIEW

**Review by enyeInc — APPROVED** [review]
LGTM

File: functions/controllers/user/getAuthUserCredentials.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+const getAuthUserCredentials = async (req, res, db) => {
+    try {
+        return getUser(req, res, db);
+    } catch (errors) {
+        return res
+            .status(INTERNAL_SERVER_ERROR)
+            .json({ message: somethingWentWrong, status: errors });
+    }
+};
```
Comment (EngrEric) [inline functions/controllers/user/getAuthUserCredentials.js]: This function is just returning the `getUser` function above.  i don't think its doing anything.

File: functions/controllers/user/index.js [hunk 2]
```diff
@@ -3,14 +3,20 @@ const { signupUser } = require('./signUp');
...
 const { getAllUser } = require('./getUsers');
+const { getAUser } = require('./getAUser');
+const { getAuthUserCredentials } = require('./getAuthUserCredentials');
 
 const login = (req, res) => loginUser(req, res, db);
 const signUp = (req, res) => signupUser(req, res, db);
 const image = (req, res) => imageUpload(req, res, db);
 const getAll = (req, res) => getAllUser(req, res, db);
+const getUserById = (req, res) => getAUser(req, res, db);
+const AuthUserCredentials = (req, res) => getAuthUserCredentials(req, res, db);
```
Comment (EngrEric) [inline functions/controllers/user/index.js:14]: this is a duplicate of the same function

File: functions/routes/index.js [hunk 3]
```diff
@@ -4,13 +4,20 @@ const router = express.Router();
...
+    AuthUserCredentials,
 } = require('../controllers/user/index');
 
 // user Routes
 router.post('/signup', signUp);
 router.post('/login', login);
 router.put('/user/image', authGuard, image);
 router.get('/users', getAll);
+router.get('/user', authGuard, AuthUserCredentials);
+router.get('/user/:userName', getUserById);
```
Comment (EngrEric) [inline functions/routes/index.js:21]: Okay i see why you added the function above.
But you can specify it by the endpoint name.
you can do something like this  `router.get('/getAuthUser', authGuard, getUserById);`

**Review by EngrEric — CHANGES_REQUESTED** [review]
Looks good, you just need to fix small duplicate of functions.

File: functions/controllers/user/getAuthUserCredentials.js (hunk 1 above)
Comment (SapeleD3) [reply functions/controllers/user/getAuthUserCredentials.js]: eslint rules stopping my functions from exceeding 20 lines so it call =s getuser and catches error so underneath 20 lines

File: functions/controllers/user/index.js (hunk 2 above)
Comment (SapeleD3) [reply functions/controllers/user/index.js:14]: no it is not .. auth user gets the users private credentials

Comment (enyeInc) [reply functions/controllers/user/index.js:14]: @SapeleD3 The pattern that you are using from line 9-14 doesn't make much sense. Can you explain why you are creating these functions this way?

Comment (SapeleD3) [reply functions/controllers/user/index.js:14]: i am breaking down the api to an MVC model making 9  - 14 the standard for each controllers minimizing code per function

**Pull request merged**